### PR TITLE
fix(anno): add annotations as contributor

### PR DESCRIPTION
**Pull request by christMent**
Allow users with Guest role to add annotations and add unit test

cm/rnd/issue_triage#482

### CODE REVIEW

**Review by waliens — CHANGES_REQUESTED** [review]
Hi @christMent, thanks for the PR ! The changes themselves seem OK to me as a solution to the issue but it seems that they have created a regression: https://gitlab.cytom.in/cm/rnd/cytomine/core/core-ce/-/pipelines/175/test_report 
Did these two failing tests succeed on your laptop ?

**Comment by christMent** [comment]
Yes they do fail, it is unrelated to this commit, is because the implementation of access rights for guest users is unfinished in the core, and we have to update multiple services and the unit tests respectively.
i will comment them out till the full implementation is done.

**Comment by waliens** [comment]
I suspect this commit changed something because these tests were passing in previous run of the pipeline anterior to the commit: https://gitlab.cytom.in/cm/rnd/cytomine/core/core-ce/-/pipelines/151/test_report

**Comment by waliens** [comment]
From what I understand, these tests fail because the minimal role for creation and deletion of UserAnnotation is set to `ROLE_USER` in test file `java/be/cytomine/authorization/ontology/UserAnnotationAuthorizationTest.java`.

```
    @Override
    protected Optional<String> minimalRoleForCreate() {
        return Optional.of("ROLE_USER");
    }

    @Override
    protected Optional<String> minimalRoleForDelete() {
        return Optional.of("ROLE_USER");
    }
```

It makes sense to me that we should update the tests to reflect that a guest can actually add and delete annotations by changing `ROLE_USER` to `ROLE_GUEST`. We should probably also handle the edition of annotation as a guest.

**Comment by christMent** [comment]
Yes, this is what I discovered as well.

But when I updated the minimalRoleForCreate to GUEST and corrected a small bug in the checking method:
`isPermissionRoleForbidden()`, it returns   `return indexCurrenRole <= indexRoleRequired;` that should return true for =
 
The other unit tests are failing because the add domain as guest is not implemented yet for these type of domain . 

For example, ImageInstanceAuthorizationTest is failing because even if we have set that 
```
   @Override
    protected Optional<String> minimalRoleForCreate() {
        return Optional.of("ROLE_GUEST");
    }
```
 at the actual service guest is not allowed (same bug as with create annotation).

So the implementation of Guest role need to finish for all domains that Guest has rights.

**Comment by waliens** [comment]
Let me rephrase to be sure I understand. What you say is that you have attempted the update to from `ROLE_USER` to `ROLE_GUEST` for 

1. user annotation creation in [UserAnnotationAuthorizationTest.java#L176](https://github.com/cytomine/Cytomine-core-spring/blob/1d96a4aec7fb6e15ff7b3d6c82df39c258c1c8c8/src/test/java/be/cytomine/authorization/ontology/UserAnnotationAuthorizationTest.java#L176),
2. user annotation deletion in [UserAnnotationAuthorizationTest.java#L181](https://github.com/cytomine/Cytomine-core-spring/blob/1d96a4aec7fb6e15ff7b3d6c82df39c258c1c8c8/src/test/java/be/cytomine/authorization/ontology/UserAnnotationAuthorizationTest.java#L181)

and nowhere else. After these two changes (+ the other fix), tests related to other domains started failing because how a guest can interact with these domains is not defined yet in the core implementation. Is it correct ?

**Comment by christMent** [comment]
Yes, I updated the methods you mentioned above, 
and I corrected :[CRDAuthorizationTest.java#L109](https://github.com/cytomine/Cytomine-core-spring/blob/1d96a4aec7fb6e15ff7b3d6c82df39c258c1c8c8/src/test/java/be/cytomine/authorization/CRDAuthorizationTest.java#L109) where the check shouldn't return true in case of equal  because then it will fail all the time for ROLE_GUEST based on this : [CRDAuthorizationTest.java#L94](https://github.com/cytomine/Cytomine-core-spring/blob/1d96a4aec7fb6e15ff7b3d6c82df39c258c1c8c8/src/test/java/be/cytomine/authorization/CRDAuthorizationTest.java#L94)

When i do that, the tests for ROLE_GUEST at UserAnnotationAuthorizationTest successfully passing 
but: 

CRU unit tests for ImageInstanceAuthorizationTest are now failing because even if minimalRoleForCreate in these test was already set to be the GUEST -> [ImageInstanceAuthorizationTest .java#L199](https://github.com/cytomine/Cytomine-core-spring/blob/1d96a4aec7fb6e15ff7b3d6c82df39c258c1c8c8/src/test/java/be/cytomine/authorization/image/ImageInstanceAuthorizationTest.java#L199) 
the current implementation is not allowing Guest to perform the action: [ImageInstanceService.java#L759](https://github.com/cytomine/Cytomine-core-spring/blob/1d96a4aec7fb6e15ff7b3d6c82df39c258c1c8c8/src/main/java/be/cytomine/service/image/ImageInstanceService.java#L759), is the same for delete and undate.

**Comment by waliens** [comment]
Ok, it seems odd that these tests "contaminate" each other but let's come back to that later.

**Review by waliens — APPROVED** [review]
